### PR TITLE
Adding alternate configuration sources

### PR DESCRIPTION
**Pull request by DavidParks8**
## I'm submitting a...

- [x] Feature

## Checklist

- [x] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
- [x] Test Included

## Description

Adding multiple configuration sources. The new behavior is first to read from the command line ng-package file, then read from the default `ng-package.json` file (if different from the command line file), then read from the `ngPackage` field from `package.json`. All configuration sources are merged using lodash in that order. This should lay the groundwork for extending multiple configs, should we want to. The primary motivation for this feature is #167.

## Does this PR introduce a breaking change?

- [x] No -> Care was taken to ensure the behavior is the same as before, if the feature is not used.

### CODE REVIEW

**Comment by DavidParks8** [comment]
I did most of the feature work, but something isn't quite right that's causing the tests to fail. At first glance, I don't see anything obviously wrong. @dherges, maybe you know what's broken.

**Comment by dherges** [comment]
Hmpf, first, can you try to `git rebase` or `git reset` your branch to the current `master` of this repo? Then, try to commit your changes in one commit and push it to the `master` of your fork. 

The build error on circle is:

```
BUILD ERROR
ENOENT: no such file or directory, open '/home/ubuntu/ng-packagr/dist/lib/conf/tsconfig.ngc.json'
Error: ENOENT: no such file or directory, open '/home/ubuntu/ng-packagr/dist/lib/conf/tsconfig.ngc.json'

module.js:473
      throw err;
      ^

Error: Cannot find module '/home/ubuntu/ng-packagr/dist/cli/ng-packagr.js'
    at Function.Module._resolveFilename (module.js:527:15)
      ^
```

So maybe a relative import path is wrong? Or the `process.cwd()` somehow stopped things from working?

**Comment by dherges** [comment]
`ng-packagr/dist/lib/conf/tsconfig.ngc.json` ... is something in the typescript compiler wrong? During the `yarn build`? Because the [`postbuild` should copy that json file to the `dist` folder](https://github.com/dherges/ng-packagr/blob/master/package.json#L79-L81)

**Comment by DavidParks8** [comment]
Please don't merge yet - I'm adding a test.

**Comment by DavidParks8** [comment]
Ready for review.

File: integration/samples/package-json/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  "description": "A sample library with configuration in package.json",
+  "version": "1.0.0-pre.0",
+  "private": true,
+  "repository": "https://github.com/dherges/ng-packagr.git",
+  "peerDependencies": {
+    "@angular/core": "^4.1.2",
+    "@angular/common": "^4.1.2"
+  },
+  "ngPackage": {
+    "$schema": "../../../src/lib/ng-package.schema.json",
```
Comment (dherges) [inline integration/samples/package-json/package.json]: Ah, do you know any possibility to extend the default schema of a `package.json`? 

Here is one ... quite old from 2013: https://www.npmjs.com/package/package.json-schema

Which schema is used for example by vscode, atom, ...? Could think of extending one of those ...

Comment (dherges) [reply integration/samples/package-json/package.json]: Here is one: http://json.schemastore.org/package

**Comment by dherges** [comment]
Double-checked the changes on a larger project and everything still :green_apple:  !

The only thing is the `$schema` for the json editing support in IDE. How can we get it to work?

schema

File: integration/samples/package-json/package.json (hunk 1 above)
Comment (dherges) [reply integration/samples/package-json/package.json]: Got it!

**Pull request merged**